src/collision/collider_backend.rs

```diff
@@ -407,51 +421,32 @@ fn update_aabb<C: AnyCollider>(
             #[cfg(feature = "2d")]
             {
                 (
-                    pos.0 + lin_vel.0 * safety_margin_factor,
-                    *rot * Rotation::radians(safety_margin_factor * ang_vel.0),
+                    pos.0
+                        + (lin_vel.0 * delta_secs)
+                            .clamp_length_max(speculative_margin.max(contact_tolerance)),
+                    *rot * Rotation::radians(ang_vel.0 * delta_secs),
                 )
             }
             #[cfg(feature = "3d")]
             {
                 let q = Quaternion::from_vec4(ang_vel.0.extend(0.0)) * rot.0;
                 let (x, y, z, w) = (
-                    rot.x + safety_margin_factor * 0.5 * q.x,
-                    rot.y + safety_margin_factor * 0.5 * q.y,
-                    rot.z + safety_margin_factor * 0.5 * q.z,
-                    rot.w + safety_margin_factor * 0.5 * q.w,
+                    rot.x + delta_secs * 0.5 * q.x,
+                    rot.y + delta_secs * 0.5 * q.y,
+                    rot.z + delta_secs * 0.5 * q.z,
+                    rot.w + delta_secs * 0.5 * q.w,
                 );
                 (
-                    pos.0 + lin_vel.0 * safety_margin_factor,
+                    pos.0
+                        + (lin_vel.0 * delta_secs)
+                            .clamp_length_max(speculative_margin.max(contact_tolerance)),
                     Quaternion::from_xyzw(x, y, z, w).normalize(),
                 )
             }
         };
-
         // Compute swept AABB, the space that the body would occupy if it was integrated for one frame
+        // TODO: Should we expand the AABB in all directions for speculative contacts?
```

Comment:
I personally like to always link tracking issues to TODO comments, but you mileage may vary